src/main/java/hanshyn/onlinebookstore/service/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+package hanshyn.onlinebookstore.service.cart;
+
+import hanshyn.onlinebookstore.dto.cart.CartItemsRequestDto;
+import hanshyn.onlinebookstore.dto.cart.CartItemsResponseDto;
+import hanshyn.onlinebookstore.dto.cart.CartItemsUpdateRequestDto;
+import hanshyn.onlinebookstore.dto.cart.ShoppingCartDto;
+import hanshyn.onlinebookstore.exception.EntityNotFoundException;
+import hanshyn.onlinebookstore.mapper.CartItemMapper;
+import hanshyn.onlinebookstore.mapper.ShoppingCartMapper;
+import hanshyn.onlinebookstore.model.Book;
+import hanshyn.onlinebookstore.model.CartItem;
+import hanshyn.onlinebookstore.model.ShoppingCart;
+import hanshyn.onlinebookstore.model.User;
+import hanshyn.onlinebookstore.repository.book.BookRepository;
+import hanshyn.onlinebookstore.repository.shopping.CartItemRepository;
+import hanshyn.onlinebookstore.repository.shopping.ShoppingCartRepository;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookRepository bookRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Authentication authentication) {
+        User user = (User) authentication.getPrincipal();
+
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(user.getId()).orElseThrow(
+                () -> new RuntimeException("Can't found user: user")
+        );
+
+        Set<CartItem> cartItem = cartItemRepository.findByShoppingCartId(shoppingCart.getId());
+
+        Set<CartItemsResponseDto> cartItemsResponseDtos = cartItem.stream()
+                .map(cartItemMapper::toDo)
+                .collect(Collectors.toSet());
+
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(shoppingCart);
+        shoppingCartDto.setCartItems(cartItemsResponseDtos);
+
+        return shoppingCartDto;
+    }
+
+    @Override
+    public CartItemsResponseDto saveBookToShoppingCart(Authentication authentication,
+                                                       CartItemsRequestDto cartItemsRequestDto) {
+        User user = (User) authentication.getPrincipal();
```

Comment:
1) For me it's better to leave work with authentication on controller level and let service work with entity required
2) Several DB modifications inside method, without proper annotation we may loose consistency of data in DB
```suggestion
    @Override
    @Transactional
    public CartItemsResponseDto saveBookToShoppingCart(User user,
                                                       CartItemsRequestDto cartItemsRequestDto) {
```